src/Mono.Android/Android.Runtime/MamXmlParser.cs

```diff
@@ -0,0 +1,159 @@
+// File must be "stand-alone"; it's included by
+// `tools/remap-mam-json-to-xml`
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Xml;
+
+using ReplacementTypesDict              = System.Collections.Generic.Dictionary<string, string>;
+using ReplacementMethodsDictStrings     = System.Collections.Generic.Dictionary<string, string>;
+using ReplacementMethodsDictStructured  = System.Collections.Generic.Dictionary<
+	(string SourceType, string SourceName, string? SourceSignature),
+	(string? TargetType, string? TargetName, string? TargetSignature, int? ParamCount, bool IsStatic)
+>;
+
+namespace Android.Runtime {
+
+	class MamXmlParser {
+
+		// https://www.unicode.org/reports/tr15/tr15-18.html#Programming%20Language%20Identifiers
+		//  <identifier> ::= <identifier_start> ( <identifier_start> | <identifier_extend> )*
+		//  <identifier_start> ::= [{Lu}{Ll}{Lt}{Lm}{Lo}{Nl}]
+		//  <identifier_extend> ::= [{Mn}{Mc}{Nd}{Pc}{Cf}]
+		//
+		// Categories which can't be part of an identifier: Cc, Me, No, Pd, Pe, Pf, Pi, Po, Ps, Sc, Sk, Sm, So, Zl, Zp, Zs
+		//
+		// Use `\t` U+0009, Category=Cc, to separate out items in ReplacementMethodsDictStrings
+
+		public static (ReplacementTypesDict ReplacementTypes, ReplacementMethodsDictStrings ReplacementMethods) ParseStrings (string xml)
+		{
+			var (types, methodsStructured) = ParseStructured (xml);
+
+			var methodsStrings  = new ReplacementMethodsDictStrings ();
+			foreach (var e in methodsStructured) {
+				var key     = $"{e.Key.SourceType}\t{e.Key.SourceName}\t{e.Key.SourceSignature}";
+				var value   = $"{e.Value.TargetType}\t{e.Value.TargetName}\t{e.Value.TargetSignature}\t{e.Value.ParamCount?.ToString () ?? ""}\t{(e.Value.IsStatic ? "true" : "false")}";
+				methodsStrings [key] = value;
+			}
+
+			return (types, methodsStrings);
```

Comment:
It might be faster to use `out` parameters for the dictionaries, to avoid instantiating a tuple

Reply:
will be moot once this is all in `libxamarin-app.so`. :-)